R/animint.R

```diff
@@ -649,12 +694,26 @@ saveLayer <- function(l, d, meta){
 saveChunks <- function(x, vars, meta){
   if(is.data.frame(x)){
     if(length(vars) == 0){
+      if(meta$geom %in% c("line", "path", "polygon", "ribbon")) {
+        if("group" %in% names(x)) {
+          # common info for groups, save only once
+          csv.name <- sprintf("%s_chunk_shape.tsv", meta$classed)
+          chunk.shape <- file.path(meta$out.dir, csv.name)
+          if(!file.exists(chunk.shape)){
+            x1 <- x[c("x", "y", "group")]
+            write.table(x1, chunk.shape, quote = FALSE, row.names = FALSE, 
+                        sep = "\t")
+          }
+          # info for each group
+          x2 <- x[, !names(x) %in% c("x", "y")]
+          x <- x2[!duplicated(x2), ]
+          }
```

Comment:
I look at the WorldBank data viz. The `color` and `clickSelects` are the same for  in each chunk tsv file. I wonder whether it's worth to reduce output chunk file size since this small redundancy make the renderer render the viz much easier. I think I should make the `geom_polyon` example work for both compiler and render, then I will extend it to WorldBank data viz.